modules/API/API.cpp

```diff
@@ -486,16 +488,18 @@ void API::init() {
         });
 
         std::string cmd_set_limit_watts = cmd_base + "set_limit_watts";
-        this->mqtt.subscribe(cmd_set_limit_watts, [&evse](const std::string& data) {
-            try {
-                const auto external_limits = get_external_limits(data, true);
-                evse->call_set_external_limits(external_limits);
-            } catch (const std::invalid_argument& e) {
-                EVLOG_warning << "Invalid limit: No conversion of given input could be performed.";
-            } catch (const std::out_of_range& e) {
-                EVLOG_warning << "Invalid limit: Out of range.";
-            }
-        });
+        this->mqtt.subscribe(
+            cmd_set_limit_watts,
+            [&evse, &r_evse_manager_energy_sink = this->r_evse_manager_energy_sink, evse_id](const std::string& data) {
+                try {
+                    const auto external_limits = get_external_limits(data, true);
+                    r_evse_manager_energy_sink.at(evse_id - 1)->call_set_external_limits(external_limits); // FIX access
+                } catch (const std::invalid_argument& e) {
+                    EVLOG_warning << "Invalid limit: No conversion of given input could be performed.";
+                } catch (const std::out_of_range& e) {
+                    EVLOG_warning << "Invalid limit: Out of range.";
```

Comment:
This kind of exception handling might a bit too weak - if there is an out of range access we can probably state, that the chargepoint is misconfigured and won't work as expected.  This should probably fail hard.